### PR TITLE
Chore(SCT-1008): Add setup for accessing secrets in the App Script Code

### PR DESCRIPTION
**Pull request by Renny4Real**
We need to be able to setup and access secrets such as the API KEY used to call the AWS S3 API Proxy.

It was suggested that we should be able to use Google's [Properties Service](https://developers.google.com/apps-script/guides/properties) to store and retrieve secrets.

From reading the documentation, it looks like adding a secret is done by calling the Properties Service API with the `setProperties` method, see [here](https://developers.google.com/apps-script/reference/properties/properties#setpropertiesproperties) and the code snippet below.

```
// Sets multiple user properties at once.
var userProperties = PropertiesService.getUserProperties();
var newProperties = {nickname: 'Bob', region: 'US', language: 'EN'};
userProperties.setProperties(newProperties);
```

If that is the case, we might need to add the file manually in the Apps Script editor with the values we need and make sure `clasp` ignores them.

TO DO after merging:
- [ ] Set secrets in Apps Script environment
- [ ] Can we test the code, take data and puts it in AWS. Should we use a temporary URL/S3 bucket setup for testing so that test data doesn't go into the production bucket?

### CODE REVIEW

File: google-scripts/process-form-submission/main.ts [hunk 1]
```diff
@@ -1,21 +1,33 @@
...
+    "REFFERALS_BUCKET_URL"
+  );
+  const S3_ENDPOINT_API_KEY =
+    PropertiesService.getScriptProperties().getProperty(
+      "REFFERALS_BUCKET_API_KEY"
+    );
+
+  const FORM_SUBMISSION_ID_COLUMN_POSITION = Number(
+    PropertiesService.getScriptProperties().getProperty("UNIQUE_ID_COLUMN_NO")
+  );
```
Comment (chrishutchinson) [inline google-scripts/process-form-submission/main.ts:17]: Should we handle a scenario where these values don't exist, or they're invalid?

File: google-scripts/process-form-submission/main.ts [hunk 2]
```diff
@@ -1,21 +1,33 @@
...
+
   var formData = event.namedValues;
```
Comment (chrishutchinson) [inline google-scripts/process-form-submission/main.ts:19]: Why are these `var` not `const`? Same everything below too

**Review by chrishutchinson — CHANGES_REQUESTED** [review]

File: google-scripts/process-form-submission/main.ts (hunk 2 above)
Comment (Renny4Real) [reply google-scripts/process-form-submission/main.ts:19]: Will change these to `const`. Wasn't sure of the difference but @JohnFarrellDev @martapederiva have helped me understand this!

File: google-scripts/process-form-submission/main.ts [hunk 3]
```diff
@@ -1,21 +1,33 @@
...
       `${REFERRALS_SHEET_NAME}`
     );
     var currentUniqueId = setUniqueIdOnSubmission(referralsSheet);
 
-    // Update object to contain its ID
+    // Update the form submission object to contain its unique ID
     formData.FormSubmissionId = [`${currentUniqueId}`];
 
+    // Send updated form submission object to AWS
     sendDataToS3();
```
[the remark this answers is not in the document]
Comment (Renny4Real) [reply google-scripts/process-form-submission/main.ts:31]: yep, can separate these functions 👍🏾

File: google-scripts/process-form-submission/__tests__/getProperties.test.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
+import {getProperties} from "../getProperties";
+import { MockPropertiesService } from "../__mocks__/google_mocks";
+
+describe.only("#getProperties", () => {
```
Comment (chrishutchinson) [inline google-scripts/process-form-submission/__tests__/getProperties.test.ts:4]:
```suggestion
describe("#getProperties", () => {
```

File: google-scripts/process-form-submission/__tests__/getProperties.test.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    expect(getProperties()).toEqual({
+      FORM_SUBMISSION_ID_COLUMN_POSITION: UNIQUE_ID_COLUMN_NO,
+      REFERRALS_SHEET_NAME: MASH_SHEET_NAME,
+      S3_ENDPOINT_API: REFERRALS_BUCKET_URL,
+      S3_ENDPOINT_API_KEY: REFERRALS_BUCKET_API_KEY,
+    });
+  });
+
+  it("should raise an error if sheet name property is empty", () => {
+    testProperties.set("MASH_SHEET_NAME", "");
```
Comment (chrishutchinson) [inline google-scripts/process-form-submission/__tests__/getProperties.test.ts:43]: Idle question, is an empty string equivalent to not having a value (aka `null`)?

Comment (chrishutchinson) [reply google-scripts/process-form-submission/__tests__/getProperties.test.ts:43]: Ah I see you have separate tests for `null` and `""`!

File: google-scripts/process-form-submission/__tests__/setUniqueIdOnSubmission.test.ts [hunk 6]
```diff
@@ -0,0 +1,125 @@
+import { setUniqueIdOnSubmission } from "../setUniqueIdOnSubmission";
+import { MockSpreadsheetApp } from "../__mocks__/google_mocks";
+
+describe("#setUniqueIdOnSubmission", () => {
+    let mockFormData: { [key: string]: string[] };
```
Comment (chrishutchinson) [inline google-scripts/process-form-submission/__tests__/setUniqueIdOnSubmission.test.ts:5]: Some of the formatting in this file, and others, looks a bit off. Can we check if we're running `prettier` on these files? If not, my recommendation is usually:

* Have `prettier` run on save in your editor
* Have a `yarn lint` job that validates the formatting is all correct as per `prettier`'s expectations (you should be able to run `prettier --check` to do this in our CI pipeline)

File: google-scripts/process-form-submission/getProperties.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
+export function getProperties() {
+  const REFERRALS_SHEET_NAME =
```
Comment (chrishutchinson) [inline google-scripts/process-form-submission/getProperties.ts:2]: Perhaps we could condense the below slightly?

```ts
const { getProperty } = PropertiesService.getScriptProperties();

const REFERRALS_SHEET_NAME = getProperty("MASH_SHEET_NAME");

// ... and so on
```

**Review by chrishutchinson — CHANGES_REQUESTED** [review]

File: google-scripts/process-form-submission/__tests__/getProperties.test.ts (hunk 5 above)
Comment (Renny4Real) [reply google-scripts/process-form-submission/__tests__/getProperties.test.ts:43]: Yeah 👍🏾

File: google-scripts/process-form-submission/__tests__/setUniqueIdOnSubmission.test.ts (hunk 6 above)
Comment (Renny4Real) [reply google-scripts/process-form-submission/__tests__/setUniqueIdOnSubmission.test.ts]: Done in 2dd6e1d

File: google-scripts/process-form-submission/getProperties.ts (hunk 7 above)
Comment (Renny4Real) [reply google-scripts/process-form-submission/getProperties.ts]: Done in 2eb2227

**Review by chrishutchinson — APPROVED** [review]

**Pull request merged**